### ISSUE TITLE
New-AzureRmADApplication

### ISSUE THREAD

**Issue by chintan3100** [Azure/azure-powershell#4044]
Not able to add active directory application using powershell script

$certificateFilePath = "D:\cert\ADTestVaultApplication.cer"
$certificate = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
$certificate.Import($certificateFilePath)
$rawCertificateData = $certificate.GetRawCertData()
$credential = [System.Convert]::ToBase64String($rawCertificateData)
$startDate= [System.DateTime]::Now
$endDate = $startDate.AddYears(1)

$credential
$adApplication = New-AzureRmADApplication -DisplayName "RahulTddddestADApplication" -IdentifierUris "http://www.rahulpnath112.com" -CertValue  $credential -StartDate $startDate -EndDate $endDate

**Error detail**

New-AzureRmADApplication : Exception of type 'System.Exception' was thrown.
At D:\cert\addcert - Copy.ps1:10 char:18
+ ... plication = New-AzureRmADApplication -DisplayName "RahulTddddestADApp ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-AzureRmADApplication], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ActiveDirectory.NewAzureADApplicationCommand

### Cmdlet(s)
New-AzureRmADApplication

### PowerShell Version
Instructions: to get PowerShell version, type `$PSVersionTable` and look for the value associated with `PSVersion`
5.*
### OS Version
Instructions: to get OS version, type `$PSversionTable` and look for value associated with `BuildVersion`

### Debug Output
Instructions: to get Debug Output, set `$DebugPreference="Continue"` and then execute the cmdlet or script causing the issue

### Script/Steps for Reproduction
$certificateFilePath = "D:\cert\ADTestVaultApplication.cer"
$certificate = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
$certificate.Import($certificateFilePath)
$rawCertificateData = $certificate.GetRawCertData()
$credential = [System.Convert]::ToBase64String($rawCertificateData)
$startDate= [System.DateTime]::Now
$endDate = $startDate.AddYears(1)

$credential
$adApplication = New-AzureRmADApplication -DisplayName "RahulTddddestADApplication" -IdentifierUris "http://www.rahulpnath112.com" -CertValue  $credential -StartDate $startDate -EndDate $endDate

**Comment by markcowl** [comment]
@chintan3100 What kind of credentials are you using?  Does the user have sufficient Graph permissions to create a new application?  Can you execute your script with ```$DebugPreference="Continue"``` to produce additional debugging information and include it here?

**Comment by chintan3100** [comment]
Thanks MarkCowl
I added $DebugPreference="Continue" . Please check above screenshot

**Comment by darshanhs90** [comment]
@chintan3100 Could you verify the end date of the certificate,since it is complaining about the credential's end date being invalid.
Also it would be great if you can send the whole debug message

**Comment by chintan3100** [comment]
Thanks Darshanhs. 
Yes look like some date formate issue.

**Comment by darshanhs90** [comment]
@chintan3100 Can we close this issue or do you need any more help regarding this issue?

**Comment by chintan3100** [comment]
Thanks for help.

**Comment by rashadrivera** [comment]
Hello All,

I was able to create apps on the June 1st, but today I get the same error that **chintan3100** raised.  I don't understand what I need to do as far as formatting dates because the argument I'm passing is of [System.DateTime] type.  

I did notice that the certificate that when I created a brand new certificate, this command worked.

**Question:**  Are we not able to upload the same certificate twice?
